src/components/molecules/ConfigurationCard/ConfigurationCard.tsx

```diff
@@ -77,10 +96,35 @@ const ConfigurationCard: React.FC<ConfigurationCardProps> = props => {
 
                 return (
                   <StyledFooterButtonsContainer>
-                    <Button onClick={onCancel} $customType="secondary" hidden={!onCancel || disabled}>
+                    <Button
+                      onClick={() => {
+                        onCancel?.();
+                        setError(null);
+                      }}
+                      $customType="secondary"
+                      hidden={!onCancel || disabled}
+                    >
                       Cancel
                     </Button>
-                    <Button onClick={onConfirm} $customType={isWarning ? 'warning' : 'primary'} disabled={disabled} hidden={!onConfirm}>
+                    <Button
+                      onClick={() => {
+                        validateFields?.()
```

Comment:
When there is no `validateFields` - shouldn't it go through `onConfirm` anyway?
```suggestion
                        Promise.resolve(validateFields?.())
```